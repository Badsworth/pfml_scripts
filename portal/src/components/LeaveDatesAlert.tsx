import Alert from "./Alert";
import React from "react";
import formatDateRange from "../utils/formatDateRange";
import { useTranslation } from "react-i18next";

interface LeaveDatesAlertProps {
  endDate?: string;
  headingLevel?: "2" | "3";
  startDate?: string;
}

/**
 * Alert component shared across multiple pages for displaying
 * a claimant's leave dates
 */
function LeaveDatesAlert(props: LeaveDatesAlertProps) {
  const { t } = useTranslation();

  // Don't render if leave dates aren't present yet
  if (!props.startDate || !props.endDate) return null;

  return (
    <Alert
      className="margin-top-2 margin-bottom-4"
      heading={t("components.leaveDatesAlert.heading")}
      headingLevel={props.headingLevel}
      headingSize="2"
      neutral
      noIcon
      state="info"
    >
      {formatDateRange(props.startDate, props.endDate)}
    </Alert>
  );
}

export default LeaveDatesAlert;
