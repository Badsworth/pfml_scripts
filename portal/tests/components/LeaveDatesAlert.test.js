import { render, screen } from "@testing-library/react";
import LeaveDatesAlert from "../../src/components/LeaveDatesAlert";
import React from "react";

jest.mock("../../src/hooks/useUniqueId");

describe("LeaveDatesAlert", () => {
  it("renders Alert with given dates", () => {
    const { container } = render(
      <LeaveDatesAlert startDate="2021-01-01" endDate="2021-01-30" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders Alert with custom headingLevel", () => {
    render(
      <LeaveDatesAlert
        startDate="2021-01-01"
        endDate="2021-01-30"
        headingLevel="3"
      />
    );
    expect(screen.getByRole("heading")).toMatchInlineSnapshot(`
      <h3
        class="usa-alert__heading font-heading-md text-bold"
        id="mock-unique-id"
      >
        Your leave dates for paid leave from PFML
      </h3>
    `);
  });

  it("with waiting period true, renders date range of wait period as expected", () => {
    render(
      <LeaveDatesAlert
        startDate="2021-01-01"
        endDate="2021-01-30"
        showWaitingDayPeriod={true}
      />
    );
    expect(
      screen.getByRole("heading", { name: "Your 7-day waiting period dates" })
    ).toBeInTheDocument();
    expect(screen.getByText("1/1/2021 to 1/7/2021")).toBeInTheDocument();
  });

  it("is empty render when one of the dates is missing", () => {
    const { container } = render(<LeaveDatesAlert startDate="2021-01-01" />);

    expect(container).toBeEmptyDOMElement();
  });

  it("renders 7 day waiting period when showWaitingDayPeriod is true", () => {
    render(
      <LeaveDatesAlert
        startDate="2021-01-31"
        endDate="2021-02-28"
        showWaitingDayPeriod
      />
    );

    expect(screen.getAllByRole("heading")).toHaveLength(2);
    expect(
      screen.getByText("Your 7-day waiting period dates").parentElement
    ).toMatchSnapshot();
  });
});
