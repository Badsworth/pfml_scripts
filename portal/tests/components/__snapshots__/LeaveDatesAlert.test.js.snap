// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaveDatesAlert renders Alert with given dates 1`] = `
<div
  class="usa-alert usa-alert--info usa-alert--no-icon c-alert--neutral margin-top-2 margin-bottom-4"
  role="region"
  tabindex="-1"
>
  <div
    class="usa-alert__body"
  >
    <h2
      class="usa-alert__heading font-heading-md text-bold"
    >
      Your leave dates for paid leave
    </h2>
    <div
      class="usa-alert__text"
    >
      1/1/2021 to 1/30/2021
    </div>
  </div>
</div>
`;
